public/app/features/dashboard-scene/scene/AlertStatesDataLayer.ts

```diff
@@ -0,0 +1,227 @@
+import { from, map, Unsubscribable, Observable } from 'rxjs';
+
+import { AlertState, AlertStateInfo, DataTopic, LoadingState, toDataFrame } from '@grafana/data';
+import { config, getBackendSrv } from '@grafana/runtime';
+import {
+  SceneDataLayerBase,
+  SceneDataLayerProvider,
+  SceneDataLayerProviderState,
+  sceneGraph,
+  SceneTimeRangeLike,
+} from '@grafana/scenes';
+import { notifyApp } from 'app/core/actions';
+import { createErrorNotification } from 'app/core/copy/appNotification';
+import { contextSrv } from 'app/core/core';
+import { getMessageFromError } from 'app/core/utils/errors';
+import { Annotation } from 'app/features/alerting/unified/utils/constants';
+import { isAlertingRule } from 'app/features/alerting/unified/utils/rules';
+import { dispatch } from 'app/store/store';
+import { AccessControlAction } from 'app/types';
+import { PromAlertingRuleState, PromRulesResponse } from 'app/types/unified-alerting-dto';
+
+import { getDashboardSceneFor } from '../utils/utils';
+
+interface AlertStatesDataLayerState extends SceneDataLayerProviderState {}
+
+export class AlertStatesDataLayer
+  extends SceneDataLayerBase<AlertStatesDataLayerState>
+  implements SceneDataLayerProvider
+{
+  private hasAlertRules = true;
+  private _timeRangeSub: Unsubscribable | undefined;
+  public topic = DataTopic.AlertStates;
+
+  public constructor(initialState: AlertStatesDataLayerState) {
+    super({
+      isEnabled: true,
+      ...initialState,
+      isHidden: true,
+    });
+  }
+
+  public onEnable(): void {
+    const timeRange = sceneGraph.getTimeRange(this);
+
+    this._timeRangeSub = timeRange.subscribeToState(() => {
+      this.runWithTimeRange(timeRange);
+    });
+  }
+
+  public onDisable(): void {
+    this._timeRangeSub?.unsubscribe();
+  }
+
+  public runLayer() {
+    const timeRange = sceneGraph.getTimeRange(this);
+    this.runWithTimeRange(timeRange);
+  }
+
+  private async runWithTimeRange(timeRange: SceneTimeRangeLike) {
+    const dashboard = getDashboardSceneFor(this);
+    const { uid, id } = dashboard.state;
+
+    if (this.querySub) {
+      this.querySub.unsubscribe();
+    }
+
+    if (!this.canWork(timeRange)) {
+      return;
+    }
+
+    let alerStatesExecution: Observable<AlertStateInfo[]> | undefined;
+
+    if (this.isUsingLegacyAlerting()) {
+      alerStatesExecution = from(
+        getBackendSrv().get(
+          '/api/alerts/states-for-dashboard',
+          {
+            dashboardId: id,
+          },
+          `dashboard-query-runner-alert-states-${id}`
+        )
+      ).pipe(map((alertStates) => alertStates));
+    } else {
+      alerStatesExecution = from(
+        getBackendSrv().get(
+          '/api/prometheus/grafana/api/v1/rules',
+          {
+            dashboard_uid: uid!,
+          },
+          `dashboard-query-runner-unified-alert-states-${id}`
```

Comment:
Can we use the dashboard `uid` here as well? 
```suggestion
          `dashboard-query-runner-unified-alert-states-${uid}`
```

Reply:
This code is just a copy of what's currently implemented in UnifiedAlertStatesWorker.